0/TransactionBuilder.js

```diff
@@ -141,6 +187,9 @@ class TransactionBuilder {
      * @returns {TransactionBuilder}
      */
     metadata(metadata) {
+        if ( this._signatures !== undefined ) {
```

Comment:
metadata can be added on the first pass only.  Since it is part of the signed content, the content can't change on the second pass, thus the return.   the way the second pass is detected is if there is a signature array in existence, which can only get created by passing in the unsigned transaction to the constructor.